File: Frontend/src/app/core/models/carriers/public-carrier-tender.dto.ts
export interface PublicCarrierTenderDto {
    assignmentId: string;
    loadNumber: string;
    customerName: string;
    origin: string;
    destination: string;
    carrierName: string;
    offeredRate: number | null;
    currency: string;
    tenderNotes?: string | null;
    tenderExpiresAt?: string | null;
    status: string;
    stops: PublicCarrierTenderStopDto[];
}

export interface PublicCarrierTenderStopDto {
    sequence: number;
    stopType: string;
    locationName: string;
    city: string;
    state: string;
    country: string;
    plannedArrivalFrom?: string | null;
    plannedArrivalTo?: string | null;
}

export interface RespondCarrierTenderDto {
    contactName?: string | null;
    contactEmail?: string | null;
    contactPhone?: string | null;
    notes?: string | null;
}

File: Backend/LogisticsPlatform.Application/DTOs/Carriers/PublicCarrierTenderDto.cs
namespace LogisticsPlatform.Application.DTOs.Carriers;

public class PublicCarrierTenderDto
{
    public Guid AssignmentId { get; set; }
    public string LoadNumber { get; set; } = string.Empty;
    public string CustomerName { get; set; } = string.Empty;
    public string Origin { get; set; } = string.Empty;
    public string Destination { get; set; } = string.Empty;
    public string CarrierName { get; set; } = string.Empty;
    public decimal? OfferedRate { get; set; }
    public string Currency { get; set; } = "USD";
    public string? TenderNotes { get; set; }
    public DateTime? TenderExpiresAt { get; set; }
    public string Status { get; set; } = string.Empty;
    public IReadOnlyList<PublicCarrierTenderStopDto> Stops { get; set; } = [];
}

public class PublicCarrierTenderStopDto
{
    public int Sequence { get; set; }
    public string StopType { get; set; } = string.Empty;
    public string LocationName { get; set; } = string.Empty;
    public string City { get; set; } = string.Empty;
    public string State { get; set; } = string.Empty;
    public string Country { get; set; } = string.Empty;
    public DateTime? PlannedArrivalFrom { get; set; }
    public DateTime? PlannedArrivalTo { get; set; }
}
